File: indexer/moulin/crawler/ar/crawler_encyclopedia.rb
#!/usr/bin/ruby

STARTTIME = Time.now

# Global defines
LANG    = "ar"
RTL     = true
PROJECT = "encyclopedia"
MASTER = "encyclopedia"
NAMESPAC= {:id => 0, :name => nil}
UNIQID  = Time.now.to_i
UNIQFD  = "/home/reg/var/#{LANG}/#{PROJECT}_#{UNIQID}"
MDWKFD  = "/var/www/reg.kiwix.org/wiki"
DBICON  = "DBI:Mysql:reg_#{LANG}_#{MASTER}"
DBIUSER = "reg"
DBIPASS = "reg"
BLOCK_SIZE	= 10485760 # 10MB

require "./#{LANG}_common.rb"

INCLUDED_NS << {:raw => "بوابة", :url => "%D8%A8%D9%88%D8%A7%D8%A8%D8%A9", :ms => "portal"}
EXCLUDED_NS << {:raw => "نقاش البوابة", :url => "%D9%86%D9%82%D8%A7%D8%B4+%D8%A7%D9%84%D8%A8%D9%88%D8%A7%D8%A8%D8%A9"}

require "../crawler.rb"

hello()

puts "Duration: #{(Time.now - STARTTIME) /60} mn."

File: indexer/moulin/crawler/fr/crawler_encyclopedia.php
#!/usr/local/bin/php
<?php

	$moulin_lang	= "fr";
	$moulin_proj	= "encyclopedia";
	$master_proj	= "encyclopedia";
	$rtl			= false;
	$namespace		= array('id' => 0, 'name' => '');
//	$block_size		= 10485760;
	
	require $moulin_lang . '_common.php';

	array_push($INCLUDED_NS, array('raw' => "Portail", 'url' => "Portail", 'ns' => 'portal'));
	array_push($INCLUDED_NS, array('raw' => "Référence", 'url' => "R%C3%A9f%C3%A9rence", 'ns' => 'reference'));

	array_push($EXCLUDED_NS, array('raw' => "Discussion Portail", 'url' => "Discussion_Portail"));
	array_push($EXCLUDED_NS, array('raw' => "Projet", 'url' => "Projet"));
	array_push($EXCLUDED_NS, array('raw' => "Discussion Projet", 'url' => "Discussion_Projet"));
	array_push($EXCLUDED_NS, array('raw' => "Discussion Référence", 'url' => "Discussion_R%C3%A9f%C3%A9rence"));

	include '../crawler.php';

?>
